### PR TITLE
Pull up clap to 4.0 series

### PR DESCRIPTION
**Pull request by imp**
Signed-off-by: Cyril Plisko <cyril.plisko@mountall.com>

## Motivation

See #1040 

## Solution

Update `Cargo.toml` and fix code accordingly.

### CODE REVIEW

File: examples/kubectl.rs [hunk 1]
```diff
@@ -21,33 +21,43 @@ use tracing::*;
...
-        Self::Pretty
+
+impl OutputMode {
+    fn as_str(&self) -> &'static str {
+        match self {
+            Self::Pretty => "pretty",
+            Self::Yaml => "yaml",
+        }
+    }
+}
```
Comment (clux) [inline examples/kubectl.rs:52]: oh, this requires more boilerplate now in clap 4 :thinking: 
i guess they want people to just use `Display`, which can be derived in other ways

**Review by clux — APPROVED** [review]
Thanks a lot!

File: examples/kubectl.rs (hunk 1 above)
Comment (imp) [reply examples/kubectl.rs:52]: Right, I can fold it into `Display` itself, but I just so used to do it that way in many other places, that fingers typed that automatically

Comment (clux) [reply examples/kubectl.rs:52]: it's fine :-)

**Pull request merged**